core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -234,6 +260,56 @@ private SampleByIntervalIterator findSampleByIntervals(
             maxTs = baseTableReader.getMaxTimestamp();
         }
 
+        // Check if we're doing incremental/full refresh on a period mat view.
+        // If so, we may need to remove the incomplete period from the refresh interval.
+        if (!rangeRefresh && viewTableMeta.periodLength > 0) {
+            TimestampSampler periodSampler = viewDefinition.getPeriodSampler();
+            if (periodSampler == null) {
+                periodSampler = TimestampSamplerFactory.getInstance(viewTableMeta.periodLength, viewTableMeta.periodLengthUnit, -1);
+                viewDefinition.setPeriodSampler(periodSampler);
+            }
+            periodSampler.setStart(viewTableMeta.timerStart);
+
+            final long delay = MatViewTimerJob.periodDelayMicros(viewTableMeta.periodDelay, viewTableMeta.periodDelayUnit);
+            long nowLocal = viewDefinition.getTimerTzRules() != null
+                    ? now + viewDefinition.getTimerTzRules().getOffset(now)
+                    : now;
+            // Period hi is exclusive, but maxTs is inclusive, so we align them.
+            final long periodHiLocal = periodSampler.round(nowLocal - delay) - 1;
+            final long periodHi = viewDefinition.getTimerTzRules() != null
+                    ? periodHiLocal - viewDefinition.getTimerTzRules().getOffset(periodHiLocal)
+                    : periodHiLocal;
+
+            // First, remove the incomplete period from both txn intervals and refresh interval.
+            if (txnIntervals != null && txnIntervals.size() > 0) {
+                txnIntervals.add(Long.MIN_VALUE, periodHi);
```

Comment:
Test coverage never go inside this if block

Reply:
Should be covered by `testImmediatePeriodMatViewWithTz()` and some other tests from `MatViewTest`